yspryn/hw6/unpack.py

```diff
@@ -0,0 +1,21 @@
+def unpack_while_loop(item):
+    res = []
+    test_list = list(item)
+    while test_list:
+        current_item = test_list.pop()
+        if isinstance(current_item, list):
+            test_list.extend(current_item)
+        else:
+            res.append(current_item)
+    return res[::-1]
```

Comment:
i think it would not work  for such cases, you will revert complete ordering:

```
unpack_while_loop([1, 2, 3])  --> [3, 2, 1]     # not something I would expect
```

Reply:
it works in such a way : 
unpack_while_loop([1, 2, 3])  --> [1, 2, 3]  
unpack_recursive([1, 2, 3]) == [1, 2, 3]